core/db_classes/EE_Registration.class.php

```diff
@@ -1955,6 +2350,15 @@ public function restore()
             $this->delete_extra_meta(EE_Registration::PRE_TRASH_REG_STATUS_KEY);
             $this->set_status($previous_status);
         }
+        $current_user = wp_get_current_user();
+        $this->add_extra_meta(
+            EE_Registration::EXTRA_META_KEY_REG_RESTORED,
+            array(
+                'restored-by' => $current_user->ID ? $current_user->display_name : 'unauthed user',
+                'timestamp'   => time(),
+                'source'      => $source,
+            )
+        );
```

Comment:
extra logging

Reply:
see my comment on `EE_Registration::delete`.  The same principle applies here (you'd just add actions for the restore path).